giftcertificate/controller/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,106 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.exceptionhandler.ValidationException;
+import com.epam.esm.service.GiftCertificateService;
+import com.epam.esm.service.dto.GiftCertificateDto;
+import com.epam.esm.service.exception.*;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.Validator;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import java.util.List;
+
+/**
+ * Class GiftCertificateController - Rest controller for process of request to GiftCertificates
+ */
+@RestController
+@RequestMapping("/api")
+public class GiftCertificateController {
+
+    private static final Logger logger = Logger.getLogger(GiftCertificateController.class);
+    /**
+     * GiftCertificateService is used for work with GiftCertificateDto
+     */
+    @Autowired
+    private GiftCertificateService service;
+
+    @Autowired
+    private Validator giftCertificateDtoValidator;
+
+    @InitBinder
+    protected void initBinder(WebDataBinder binder) {
+        binder.setValidator(giftCertificateDtoValidator);
+    }
+
+    /**
+     * @param sortType  - it is name of field in table gitf_certificates of DB
+     * @param orderType ASC or DESC
+     * @return List<GiftCertificateDto>
+     * @throws RequestParamServiceException if params don't correlate with name of field in DB
+     *                                      method readAll - read all GiftCertificates fro DB
+     */
+    @GetMapping("/gift_certificates")
```

Comment:
It s not a good practice to use underscores as part of URL for API. Try to avoid the word delimiter when you can do without it.  Better to use hyphens as delimiter